torchao/experimental/ops/mps/register.mm

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the license found in the
+// LICENSE file in the root directory of this source tree.
+
+// clang-format off
+#include <torch/extension.h>
+#include <ATen/native/mps/OperationUtils.h>
+#include <torchao/experimental/kernels/mps/src/lowbit.h>
+// clang-format on
+
+namespace torchao {
+namespace kernels {
+namespace mps {
+namespace lowbit {
+namespace aten {
+
+using Tensor = at::Tensor;
+using namespace at::native::mps;
+
+// LowBit Quantized Linear on MPS Backend
+template <int nbit>
+Tensor linear_mps_kernel(
+    const Tensor& A,
+    const Tensor& B,
+    int64_t group_size,
+    const Tensor& SZ) {
+  auto M = A.size(0);
+  auto N = B.size(0);
+  auto K = A.size(1);
+  auto C = at::empty({M, N}, A.options());
+
+  LowBitQuantWeights<nbit>::linear(
+      getMTLBufferStorage(A),
```

Comment:
Do we not need to assert that tensor is already on metal device?

For example if subsequent kernel is on cpu, then it will assume that the data is already available in the buffer pointed by the pointer. But depending on whether the work is submitted _and_ finished on GPU this may or may not be true.

e.g. Here C is created on same device as A.